### PR TITLE
Fix slow paged room list regression

### PR DESCRIPTION
**Pull request by BillCarsonFr**
## Type of change

- [x] Bugfix

## Content

A new parameter was added to resolve the list of parent summaries when loading the rooms paged list.
This param is affecting performances a lot when you have some decent space hierarchy. And it was called unproperly when displaying the regular room list (should only be needed for search/filter rooms).

Also the new added `flattenParents` field (that was slowing the queries) was never used. Instead the space parent was used. I think improperly because it only reflect the presence of a `m.space.parent`relation, that is not used to decide hierarchy (`m.space.child`) are used for that.

@ericdecanini  can you confirm?

See https://github.com/vector-im/element-android/issues/3830 regarding recent degradation of performances.

## Tested devices

- [x] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt
Comment (ericdecanini) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt:237]:
```suggestion
            getFlattenParents: Boolean = false, // Heavily impacts performance when true. Only use when space parent is absolutely required
```

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt
Comment (ericdecanini) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt:231]: Feels weird to have a partial java doc covering only one parameter. I would resolve this TODO and complete the javadoc here (and preferably for the method above too, to avoid confusion later)
```suggestion
     * Gets filtered page of room summaries so that [insert use of filtered here]
     *
     * @param queryParams - tba
     * @param pagedListConfig - tba
     * @param sortOrder - tba
     * @param getFlattenParents When true, the list of known parent and grand parent summaries will be resolved.
```

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomSummaryItemFactory.kt
Comment (ericdecanini) [inline vector/src/main/java/im/vector/app/features/home/room/list/RoomSummaryItemFactory.kt:212]: Although this is nice, I'm thinking now to maintain parity with web, we only take the last element in `flattenParents` for the subtitle. If design would deem it's better to have all parents separated by comma, we can have a meta ticket to change that across all platforms

**Review by ericdecanini — CHANGES_REQUESTED** [review]
I think the doc change is very much needed. There's a similar method in `RoomService` called `getPagedRoomSummariesLive` that feels like the default method used while `getFilteredPagedRoomSummariesLive` sounds like it's only used in `FilteredRoomsActivity`. This issue highlights that this isn't the case, and it would be good to make that clearer in the code through these java docs

File: vector/src/main/java/im/vector/app/features/home/room/list/RoomSummaryItemFactory.kt
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/home/room/list/RoomSummaryItemFactory.kt:212]: Doesn't look to me what web is taking the last one only. What are the requirement exactly?
Can you take over it as I believe you are the one that initialy developed it?

**Comment by fedrunov** [comment]
@BillCarsonFr were you able to reproduce or you just assume that flattened parents is the cause? I've tested on many spaces and there were no problems with db read, so if you can reproduce, could you share your setup?

Also, issue was created before flatten parents flag was added, so I'm not sure this actually fixes problem

**Comment by BillCarsonFr** [comment]
> @BillCarsonFr were you able to reproduce or you just assume that flattened parents is the cause? I've tested with many spaces and there were no problems with db read, so if you can reproduce, could you share your setup?
> 
> Also, issue was created before flatten parents flag was added ([PR](https://github.com/vector-im/element-android/pull/5860)). So I'm not sure this actually fixes problem

I can reproduce 100% with my personal account.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt:237]: Good point, but please add this mention in the Kdoc and not here, or it will not be exported.

**Review by bmarty — COMMENTED** [review]
Please add a file for the changelog

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/RoomService.kt]: If you use `tba` you should keep the `TODO`

**Review by bmarty — APPROVED** [review]
LGTM after a static review.
Please do not merge on develop, I will create a hotfix, as it's blocking the release 1.4.18.

**Comment by bmarty** [comment]
All the 4 commits have been cherry-picked to create the hotfix 1.4.19. The changes are now on `develop` and on `main`.

**Comment by waclaw66** [comment]
Another regression https://github.com/vector-im/element-android/issues/6272 that didn't make it to 1.4.19...

**Pull request closed without merging**
